src/track/beatgrid.cpp

```diff
@@ -23,72 +23,76 @@ namespace mixxx {
 
 class BeatGridIterator : public BeatIterator {
   public:
-    BeatGridIterator(double dBeatLength, double dFirstBeat, double dEndSample)
-            : m_dBeatLength(dBeatLength),
-              m_dCurrentSample(dFirstBeat),
-              m_dEndSample(dEndSample) {
+    BeatGridIterator(audio::FrameDiff_t beatLengthFrames,
+            audio::FramePos firstBeatPosition,
+            audio::FramePos endPosition)
+            : m_beatLengthFrames(beatLengthFrames),
+              m_endPosition(endPosition),
+              m_currentPosition(firstBeatPosition) {
     }
 
     bool hasNext() const override {
-        return m_dBeatLength > 0 && m_dCurrentSample <= m_dEndSample;
+        return m_beatLengthFrames > 0 && m_currentPosition <= m_endPosition;
     }
 
-    double next() override {
-        double beat = m_dCurrentSample;
-        m_dCurrentSample += m_dBeatLength;
-        return beat;
+    audio::FramePos next() override {
+        const audio::FramePos beatPosition = m_currentPosition;
```

Comment:
Use the opportunity to simplify this code: return m_currentPosition + m_beatLengthFrames

Reply:
no, then the iterator would not advance and always return the same position.

Reply:
Sorry, confused the variables.